diff --git a/src/interfaces/libpq/fe-secure-openssl.c b/src/interfaces/libpq/fe-secure-openssl.c
index ffbade3..870b47d 100644
--- a/src/interfaces/libpq/fe-secure-openssl.c
+++ b/src/interfaces/libpq/fe-secure-openssl.c
@@ -1544,12 +1544,18 @@ pgtls_open_client(PGconn *conn)
 							 * codes listed here, but a hint about protocol
 							 * versions seems like it's appropriate for all.
 							 */
+#ifdef SSL_R_NO_PROTOCOLS_AVAILABLE
 						case SSL_R_NO_PROTOCOLS_AVAILABLE:
+#endif
 						case SSL_R_UNSUPPORTED_PROTOCOL:
+#ifdef SSL_R_BAD_PROTOCOL_VERSION_NUMBER
 						case SSL_R_BAD_PROTOCOL_VERSION_NUMBER:
+#endif
 						case SSL_R_UNKNOWN_PROTOCOL:
 						case SSL_R_UNKNOWN_SSL_VERSION:
+#ifdef SSL_R_UNSUPPORTED_SSL_VERSION
 						case SSL_R_UNSUPPORTED_SSL_VERSION:
+#endif
 						case SSL_R_WRONG_SSL_VERSION:
 						case SSL_R_WRONG_VERSION_NUMBER:
 						case SSL_R_TLSV1_ALERT_PROTOCOL_VERSION:
@@ -1895,6 +1901,44 @@ my_sock_write(BIO *h, const char *buf, int size)
 	return res;
 }
 
+static int
+my_bio_create(BIO *bio)
+{
+	BIO_set_data(bio, NULL);
+	BIO_set_init(bio, 1);
+	return 1;
+}
+
+static int
+my_bio_destroy(BIO *bio)
+{
+	if (!bio)
+		return 0;
+	BIO_set_data(bio, NULL);
+	BIO_set_init(bio, 0);
+	return 1;
+}
+
+static long
+my_bio_ctrl(BIO *bio, int cmd, long num, void *ptr)
+{
+	(void) ptr;
+
+	switch (cmd)
+	{
+		case BIO_CTRL_GET_CLOSE:
+			return BIO_get_shutdown(bio);
+		case BIO_CTRL_SET_CLOSE:
+			BIO_set_shutdown(bio, (int) num);
+			return 1;
+		case BIO_CTRL_FLUSH:
+		case BIO_CTRL_DUP:
+			return 1;
+		default:
+			return 0;
+	}
+}
+
 static BIO_METHOD *
 my_BIO_s_socket(void)
 {
@@ -1928,14 +1974,11 @@ my_BIO_s_socket(void)
 		 * As of this writing, these functions never fail. But check anyway,
 		 * like OpenSSL's own examples do.
 		 */
 		if (!BIO_meth_set_write(res, my_sock_write) ||
 			!BIO_meth_set_read(res, my_sock_read) ||
-			!BIO_meth_set_gets(res, BIO_meth_get_gets(biom)) ||
-			!BIO_meth_set_puts(res, BIO_meth_get_puts(biom)) ||
-			!BIO_meth_set_ctrl(res, BIO_meth_get_ctrl(biom)) ||
-			!BIO_meth_set_create(res, BIO_meth_get_create(biom)) ||
-			!BIO_meth_set_destroy(res, BIO_meth_get_destroy(biom)) ||
-			!BIO_meth_set_callback_ctrl(res, BIO_meth_get_callback_ctrl(biom)))
+			!BIO_meth_set_ctrl(res, my_bio_ctrl) ||
+			!BIO_meth_set_create(res, my_bio_create) ||
+			!BIO_meth_set_destroy(res, my_bio_destroy))
 		{
 			goto err;
 		}
